platform/SingletonPtr.h

```diff
@@ -129,14 +134,48 @@ struct SingletonPtr {
         return *get();
     }
 
-    // This is zero initialized when in global scope
-    mutable void *_ptr;
+    /** Get a pointer to the underlying singleton
+     *
+     * Gets a pointer without initialization - can be
+     * used as an optimization when it is known that
+     * initialization must have already occurred.
+     *
+     * @returns
+     *   A pointer to the singleton, or NULL if not
+     *   initialized.
+     */
+    T *get_no_init() const
```

Comment:
I think not here - I think returning NULL if it's not initialised is potentially useful API. You might want to perform some operation if it's active, but not start it.

Callers could maybe assert, but I don't think asserting ahead of a NULL dereference is that useful either - it will go bang with a pretty easy to debug backtrace.